server/journeys/record-investigation/triggers/schemas.ts

```diff
@@ -1,14 +1,17 @@
 import z from 'zod'
+import { Request } from 'express'
 import { createSchema } from '../../../middleware/validationMiddleware'
 
 const REASONS_MSG = 'Enter a description of the prisoner’s triggers'
 const TOO_LONG_ERROR_MSG = 'Description of the prisoner’s triggers must be 4,000 characters or less'
 
-export const schema = createSchema({
-  personsTrigger: z
-    .string({ message: REASONS_MSG })
-    .max(4000, TOO_LONG_ERROR_MSG)
-    .refine(value => value?.trim().length > 0, { message: REASONS_MSG }),
-})
+export const schemaFactory = async (_req: Request) => {
+  return createSchema({
+    personsTrigger: z
+      .string({ message: REASONS_MSG })
+      .max(4000, TOO_LONG_ERROR_MSG)
+      .refine(value => value?.trim().length > 0, { message: REASONS_MSG }),
+  })
+}
```

Comment:
this can just be a simple schema, the SchemaFactory type is created to handle validation rules that need to be built from a req (eg, when we need to read Reference data from api)